### PR TITLE
Common student errors page

### PR DESCRIPTION
**Pull request by matgar1997**
This PR contains a couple of images and some documentation that can be used to help students debug error on homework submissions

### CODE REVIEW

File: _docs/student/student_errors.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+
+### Common user errors
+
+Your code is not tested locally first, Submitty should be used as the
+final submission server, not as a debugger. Before submitting anything
+make sure that your able to run your code locally without error.
+In addition the instructor may deciede to use hidden tests which still
+reveal the grade. Often these tests corner cases and exactly what those
+could be vary by your program but common cases to consider are positive,
```
Comment (zjxiaohan) [inline _docs/student/student_errors.md]:
```suggestion
could vary by your program but common cases to consider are positive, 
```

File: _docs/student/student_errors.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+### Common user errors
+
+Your code is not tested locally first, Submitty should be used as the
+final submission server, not as a debugger. Before submitting anything
+make sure that your able to run your code locally without error.
+In addition the instructor may deciede to use hidden tests which still
+reveal the grade. Often these tests corner cases and exactly what those
+could be vary by your program but common cases to consider are positive, 
+negative, zero, and floating point values and how your code performs give
+unexpected but valid input.
```
Comment (zjxiaohan) [inline _docs/student/student_errors.md]: I am not sure if we really want to be so specific about what the corner cases are, as some programs may be unrelated to number operations. I am afraid that these hints may confuse someone working with some completely different problems (like assignments on string operations).

File: _docs/student/student_errors.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+Often times there may be a missing punctuation or small typo in the output
+If you see that your not getting full cedit carefully read over the student
```
Comment (zjxiaohan) [inline _docs/student/student_errors.md]:
```suggestion
If you see that you are not getting full credit carefully read over the student
```

File: _docs/student/student_errors.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+and expected output to be sure you didnt miss a red highlighting mark
```
Comment (zjxiaohan) [inline _docs/student/student_errors.md]:
```suggestion
and expected output to be sure you didn't miss a red highlighting mark
```

File: _docs/student/student_errors.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+![](/images/typo.png)
+
+Invisible typos are hard to see but with the visualize whitespace button
```
Comment (zjxiaohan) [inline _docs/student/student_errors.md]:
```suggestion
Invisible typos are hard to see but the visualize whitespace button
```

File: _docs/student/student_errors.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+next to the output that can make it easier to catch pesky spaces and new lines
```
Comment (zjxiaohan) [inline _docs/student/student_errors.md]:
```suggestion
next to the output can make it easier to catch pesky spaces and new lines
```

File: _docs/student/student_errors.md [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+
+One of the most common errors that can occur happen because Submitty runs on
+Ubuntu, a version on Linux, and many students use Windows. If your running into
```
Comment (zjxiaohan) [inline _docs/student/student_errors.md]:
```suggestion
Ubuntu, a Linux distribution, and many students use Windows. If you are running into
```

File: _docs/student/student_errors.md [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+an issue with new lines. That is probably due to "\r\n" vs "\n". Windows uses
+"\r\n" and if your output uses that for new lines your output wont match Submitty.
```
Comment (zjxiaohan) [inline _docs/student/student_errors.md]:
```suggestion
"\r\n" and if your output uses that for new lines your output won't match Submitty. 
```

File: _docs/student/student_errors.md [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+next to the output that can make it easier to catch pesky spaces and new lines
+
+One of the most common errors that can occur happen because Submitty runs on
+Ubuntu, a version on Linux, and many students use Windows. If your running into
+an issue with new lines. That is probably due to "\r\n" vs "\n". Windows uses
+"\r\n" and if your output uses that for new lines your output wont match Submitty. 
+
+### Kill Signal
+
+Kill signals exist to prevent a piece of code from running to long and using
```
Comment (zjxiaohan) [inline _docs/student/student_errors.md]:
```suggestion
Kill signals exist to prevent a piece of code from running too long and using
```

File: _docs/student/student_errors.md [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+up reasources. Sometimes a kill signal will be sent due to an infinte loop
+in your code. If you do not get an infinte loop during local testing it is
+possible the the inputs being used on submitty are causing your code to loop.
```
Comment (zjxiaohan) [inline _docs/student/student_errors.md]:
```suggestion
possible the inputs being used on Submitty are causing your code to loop.
```

File: _docs/student/student_errors.md [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+
+One of the most common errors that can occur happen because Submitty runs on
+Ubuntu, a version on Linux, and many students use Windows. If your running into
+an issue with new lines. That is probably due to "\r\n" vs "\n". Windows uses
+"\r\n" and if your output uses that for new lines your output wont match Submitty. 
+
+### Kill Signal
+
+Kill signals exist to prevent a piece of code from running to long and using
+up reasources. Sometimes a kill signal will be sent due to an infinte loop
```
Comment (zjxiaohan) [inline _docs/student/student_errors.md]:
```suggestion
up resources. Sometimes a kill signal will be sent due to an infinite loop
```

File: _docs/student/student_errors.md [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+in your code. If you do not get an infinte loop during local testing it is
+possible the the inputs being used on submitty are causing your code to loop.
+Even if you did not experiance this during testing
```
Comment (zjxiaohan) [inline _docs/student/student_errors.md]:
```suggestion
Even if you did not experience this during testing.
```
